bftengine/src/bftengine/ReplicaImp.hpp

```diff
@@ -480,6 +483,9 @@ class ReplicaImp : public InternalReplicaApi, public ReplicaForStateTransfer {
   void startConsensusProcess(PrePrepareMsg* pp);
   bool isSeqNumToStopAt(SeqNum seq_num);
 
+  bool validatePrePrepareMsg(const PrePrepareMsg* msg, const bftEngine::impl::ReplicasInfo& replicasInfo);
+  bool validatePreProcessedResultSignatures(preprocessor::PreProcessedResultMsg& msg);
```

Comment:
As this is validate [read only] method, should msg be const& ?

Reply:
Good point. I tried this while writing the function. There is a call to msg.getResultSignaturesBuf() in it:
`const auto [buf, buf_len] = msg.getResultSignaturesBuf();`
getResultSignaturesBuf returns pointer so it's not correct to be const. This was the only reason to  get non-const reference in validatePreProcessedResultSignatures().